app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -247,9 +252,25 @@ def find_gpg_key
     end
 
     def repository_params
-      keys = [:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name]
-      keys += [:label, :content_type] if params[:action] == "create"
-      params.require(:repository).permit(*keys)
+      if params[:action] == "create"
+        params.require(:repository).permit(:label,
+                                           :content_type,
+                                           :url,
+                                           :gpg_key_id,
+                                           :unprotected,
+                                           :name,
+                                           :checksum_type,
+                                           :docker_upstream_name,
+                                           :ostree_branches => [])
```

Comment:
Why did you break apart the array here into two separate instances of calling `permit`?